docs/blog/posts/argocd-keptn-health.md

```diff
@@ -0,0 +1,163 @@
+due to an internal misconfiguration of the application.
+
+We will deploy a simple podtato-head application, which consists of multiple
+Deployments and Services via ArgoCD.
+The Argo Application deploying the manifests can look like the following:
+
+```yaml
+{% include "./argocd-keptn-health/argo-app.yaml" %}
+```
+
+After a few moments, the `podtato-head` application is successfully deployed and all pods
+are running.
+
+![Running Pods](./argocd-keptn-health/running-pods.png)
+
+We can also examine the ArgoCD UI and everything seems to be working as expected and the
+`podtato-head` application is healthy.
+
+![Healthy App](./argocd-keptn-health/healthy-app.png)
+
+Let's now try to add some health checks of the `podtato-head` application
+and use Keptn to execute them.
+For this we are going to use the
+[Keptn Release Lifecycle Management](https://keptn.sh/stable/docs/getting-started/lifecycle-management/)
+feature and perform the checks via `KeptnTasks`.
+
+First we need to add `KeptnTaskDefinition` into our GitOps repository, where our
+`podtato-head` application lives.
+It defines a simple test of reachability of one of the `podtato-head` application
+services and confirms that this service is available.
+For simplicity we choose a very simple task that will fail for example purposes,
+but in a real deployment more advanced checks can be used.
+
+```yaml
+{% include "./argocd-keptn-health/taskdefinition.yaml" %}
+```
+
+Additionally, we need to annotate the `podtato-head-frontend` Deployment to execute
+the task as part of `post-deployment` checks.
+
+```yaml
+{% include "./argocd-keptn-health/annotation.yaml" %}
+```
+
+After these two changes are made in our GitOps repository, we can restart the deployment
+of `podtato-head` application.
+After the restart, ArgoCD will re-deploy the application, Keptn waits until all of the
+application pods are running and afterwards executes `post-deployment` tasks.
+
+Due to misconfiguration of the `podtato-head-frontend` Deployment, the service is exposed on
+port `8080` instead of `8081` as it's expected by the task.
+The executed `KeptnTask` therefore fails.
+
+Here we can see that with the use of Keptn we can perform more advanced health checks
+(tasks or evaluations) and verify that the application is healthy during the process
+of deployment which is performed by ArgoCD.
+Similary to `KeptnTasks`, `KeptnEvaluations` can be executed as part of a quality health checks
+for our application.
+
+## How to show Keptn health status in ArgoCD UI?
+
+Using Keptn together with ArgoCD brings a lot of value, which we saw in the previous section,
+but looking on application health status in the cluster is not the best user experience.
+The data should be nicely displayed in the ArgoCD UI to provide the user with an overview
+if the application was deployed, if it's synchronized and if it's healthy, all in
+one place.
+
+Due to this reason, we are going to enhance ArgoCD UI with additional application health
```

Comment:
this sounds a bit like we are modifying the actual ArgoCD UI - further above it was mentioned that this will be a UI extension, so let's briefly mention this here as well to avoid confusion - something like 

```suggestion
Due to this reason, we are going to implement an ArgoCD UI extension with additional application health
```